mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobDefinition.java

```diff
@@ -141,6 +151,16 @@ public void validateSchedulingInfo() throws InvalidJobException {
         validateSchedulingInfo(false);
     }
 
+    private void postProcess() {
+        if (this.deploymentStrategy != null && !Strings.isNullOrEmpty(this.deploymentStrategy.getResourceClusterId())) {
+            this.labels.put(
+                SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label,
+                new Label(
+                    SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label,
+                    this.deploymentStrategy.getResourceClusterId()));
+        }
```

Comment:
What's the rationale for having this information in both DeploymentStrategy as well as in the form of labels?

Reply:
IMO the system label is more of a impl detail and having this leaked to public API is putting too much logic to the sender/UI side.